src/systems/filecoin_blockchain/chainsync/_index.md

```diff
@@ -1,58 +1,388 @@
+MAY opt to use additional protocols to achieve the results.
+That said, Nodes MUST implement the version of `ChainSync` as described in this spec in order to
+be considered implementations of Filecoin. Test suites will assume this protocol.
+
+# ChainSync State Machine
+
+`ChainSync` uses the following _conceptual_ state machine. Since this is a _conceptual_ state machine,
+implementations MAY deviate from implementing precisely these states, or dividing them strictly.
+Implementations MAY blur the lines between the states. If so, implementations MUST ensure security
+of the altered protocol.
+
+State Machine:
+
+{{< diagram src="chainsync_fsm.dot.svg" title="ChainSync State Machine" >}}
+
+## Details for all states:
+
+### Block Fetching and Validation
+
+- `ChainSync` selects and maintains a set of the most likely heads to be correct from among those received
+  via `BlockPubsub`. As more blocks are received, the set of `BestHeads` is reevaluated.
+- `ChainSync` fetches `Blocks`, `Messages`, and `StateTree` through the `Graphsync` protocol.
+- `ChainSync` maintains sets of `Blocks/Tipsets` in `Graphs` (see `ChainSync.id`)
+- `ChainSync` gathers a list of `LatestTipsets` from `BlockPubsub`, sorted by likelihood of being the best chain (see below).
+- `ChainSync` makes requests for chains of `BlockHeaders` to close gaps between  `LatestTipsets`
+- `ChainSync` forms partial unvalidated chains of `BlockHeaders`, from those received via `BlockPubsub`, and those requested via `Graphsync`.
+- `ChainSync` attempts to form fully connected chains of `BlockHeaders`, parting from `StateTree`, toward observed `Heads`
+- `ChainSync` minimizes resource expenditures to fetch and validate blocks, to protect against DOS attack vectors.
+  `ChainSync` employs **Progressive Block Validation**, validating different facets at different stages of syncing.
+- `ChainSync` delays syncing `Messages` until they are needed. Much of the structure of the partial chains can
+  be checked and used to make syncing decisions without fetching the `Messages`.
+
+### Progressive Block Validation
+
+- Blocks can be validated in progressive stages, in order to minimize resource expenditure.
+- Validation computation is considerable, and a serious DOS attack vector.
+- Secure implementations must carefully schedule validation and minimize the work done by pruning blocks without validating them fully.
+- **Progressive Stages of Block Validation**
+  - _(TODO: move this to blockchain/Block section)_
+  - **BV0 - Syntactic Validation**: Validate data structure packing and ensure correct typing.
+  - **BV1 - Light Consensus State Checks**: Validate `b.ChainWeight`, `b.ChainEpoch`, `b.MinerAddress`, are plausible (some ranges of bad values can be detected easily, especially if we have the state of the chain at `b.ChainEpoch - consensus.LookbackParameter`. Eg Weight and Epoch have well defined valid ranges, and `b.MinerAddress`
```

Comment:
does this or does this not depend on having some amount of chain history linking to this block?